common/types/test_util.go

```diff
@@ -0,0 +1,231 @@
+package types
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/uber/cadence/common/testing/testdatagen"
+)
+
+type byteSizer interface{ ByteSize() uint64 }
+
+// AssertByteSizeMatchesReflect verifies v.(byteSizer).ByteSize()
+// matches an independent reflective deep-size computation.
+func AssertByteSizeMatchesReflect(t *testing.T, v any) {
```

Comment:
should the `v any` be cache.Sizable`? I'm not sure I understand to what extent it's useful to pass non-sized stuff in?